src/Cli/MergeConfig.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Buffers;
+using System.Text;
+using System.Text.Json;
+
+namespace Cli
+{
+    /// <summary>
+    /// This Class contains methods to merge config files based on DAB_ENVIRONMENT value
+    /// to give a similar experience to that of dotnet development.
+    /// </summary>
+    public class MergeConfig
+    {
+        /// <summary>
+        /// Parse the string and call the helper methods to merge based on their value kind, i.e. Objects or Array.
+        /// </summary>
+        public static string Merge(string originalJson, string newContent)
+        {
+            ArrayBufferWriter<byte> outputBuffer = new();
+
+            using (JsonDocument jDoc1 = JsonDocument.Parse(originalJson))
```

Comment:
nit: Rather using name like jDoc1 or jDoc2, we can use names related to the original/overriding json. Looks better!